plugin/pkg/shoot/dns/admission.go

```diff
@@ -301,7 +309,15 @@ func assignDefaultDomainIfNeeded(shoot *core.Shoot, projectLister corelisters.Pr
 			if shoot.Spec.DNS == nil {
 				shoot.Spec.DNS = &core.DNS{}
 			}
-			generatedDomain := fmt.Sprintf("%s.%s.%s", shoot.Name, project.Name, domain)
+			shootDNSName := shoot.Name
+			if len(shoot.Name) == 0 && len(shoot.GenerateName) > 0 {
+				random, err := gardenerutils.GenerateRandomStringFromCharset(15, "0123456789abcdefghijklmnopqrstuvwxyz")
```

Comment:
Why "15"? Shouldn't it be `len(shoot.GenerateName)+5`?

Reply:
No specific calculation to reach this number. We can reduce or increase. I am not aware of any restriction on DNS subdomain. I picked it at random but reasonably safe. Just didn't thought too long random dns name is not needed. but too small will limit the range of unique DNS name.

Generatename and DNS subdomain don't have dependency anyways. So i didn't get why do you expect this to dependent on len(GenerateName)?

Reply:
> Generatename and DNS subdomain don't have dependency anyways. So i didn't get why do you expect this to dependent on len(GenerateName)?

No specific reason, it doesn't matter too much. It just feels more natural to have the length similar to the length of the shoot name instead of arbitrarily choosing `15`